### ISSUE TITLE
[BUG] Category page and Shipping Details - use SfButton instead of role="button"

### ISSUE THREAD

**Issue by justyna-13** [vuestorefront/storefront-ui#2220]
**Describe the bug**
There are SfIcon components that have `role="button"` in Category page and SfShippingDetails template. These  `role="button"` should be removed and SfIcon components should be wrapped in SfButton component. SfButton should have `sf-button--pure` class 

**How to reproduce**
1. Go to Category page / SfShippingDetails
2. Scroll down to SfIcon with `role="button"`

**Expected behavior**
SfIcon is wrapped in SfButton. Similar examples are in SHeader

**Actual behavior**
SfIcon has role button

<img width="239" alt="Screenshot 2022-01-26 at 14 25 51" src="https://user-images.githubusercontent.com/46591755/151171141-200f15df-f5a4-4fb7-bcc7-3a4ea68a5578.png">
<img width="270" alt="Screenshot 2022-01-26 at 14 26 19" src="https://user-images.githubusercontent.com/46591755/151171175-2b86e8f1-59d1-436a-b769-cc1d30c6e8ce.png">

**Comment by shatanov** [comment]
Hi, I wanted to take this task)

**Comment by justyna-13** [comment]
Sure @shatanov :) it's yours
